src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -2090,6 +2090,73 @@ def _process_derived_type_decln(self, parent, decl, visibility_map):
             tsymbol.datatype = UnknownFortranType(str(decl))
             tsymbol.interface = UnknownInterface()
 
+    def _get_partial_datatype(self, node, parent, visibility_map):
+        '''Try to obtain partial datatype information from node by removing
+        any unsupported properties in the declaration. This method
+        assumes that the provided declaration statement only declares
+        a single variable.
+
+        :param node: fparser2 node containing the declaration statement.
+        :type node: :py:class:`fparser.two.Fortran2008.Type_Declaration_Stmt`
+            or :py:class:`fparser.two.Fortran2003.Type_Declaration_Stmt`
+        :param parent: PSyIR node in which to insert the symbols found.
+        :type parent: :py:class:`psyclone.psyir.nodes.Node`
```

Comment:
Would scope be a better name for this arg?

Reply:
Changed.